src/main/kotlin/pt/up/fe/ni/website/backend/controller/AccountController.kt

```diff
@@ -1,23 +1,32 @@
 package pt.up.fe.ni.website.backend.controller
 
+import org.springframework.validation.annotation.Validated
 import org.springframework.web.bind.annotation.GetMapping
 import org.springframework.web.bind.annotation.PathVariable
 import org.springframework.web.bind.annotation.PostMapping
-import org.springframework.web.bind.annotation.RequestBody
 import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RequestPart
 import org.springframework.web.bind.annotation.RestController
+import org.springframework.web.multipart.MultipartFile
+import pt.up.fe.ni.website.backend.annotations.validation.ValidImage
 import pt.up.fe.ni.website.backend.dto.entity.AccountDto
+import pt.up.fe.ni.website.backend.model.Account
 import pt.up.fe.ni.website.backend.service.AccountService
 
 @RestController
 @RequestMapping("/accounts")
+@Validated
 class AccountController(private val service: AccountService) {
     @GetMapping
     fun getAllAccounts() = service.getAllAccounts()
 
     @GetMapping("/{id}")
     fun getAccountById(@PathVariable id: Long) = service.getAccountById(id)
 
-    @PostMapping("/new")
-    fun createAccount(@RequestBody dto: AccountDto) = service.createAccount(dto)
+    @PostMapping("/new", consumes = ["multipart/form-data"])
+    fun createAccount(@RequestPart account: AccountDto, @RequestParam @ValidImage photo: MultipartFile?): Account {
```

Comment:
#108 needs to implement this as well in edit

Reply:
Do you think `account` is a better name than `dto`? dto seems less confusing to me

Reply:
Seems good